### DISCUSSION TITLE
In APIRouter Staticfile is not working..

### DISCUSSION THREAD

**Discussion by sindhubha** [fastapi/fastapi#9574] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
├── app
│   ├
│   ├── main.py
│   ├
│   └── routers
│   │   ├
│   │   ├── api.py
│   │    
│   └── src
│       ├── endpoints|--attachments-----------|--company |-- company_logo
        |            |-- master_company       |          |-- group_logo
        |            |-- master_department    |
        |            |-- master_branch
        |            
        |-- models|--mssql
                  |
                  |--mysql|--master_company_model
                          |--master_branch_model
                          |--master_department_model
```

### Description

This is the structure of our project source .

static file serving works when the file is located in the main application directory, but it doesn't work when the file is located in a router (master_company) subdirectory 

### Operating System

Windows

### FastAPI Version

0.95.1

### Python Version

3.9.12

**Comment by WilliamStam** [comment, marked as answer]
very stupid question. are you linking to the right folder? the folder structure will be from wherever your "main" entry point is if im reading your table correctly it should be something like `app.mount("/static", StaticFiles(directory="src/endpoints"), name="static")` 

you could also create a normal route to bypass the static handler thing as well to test it with or something

```
from fastapi.responses import FileResponse, Response
@route.get("/static/{rest_of_path:path}", response_class=Response, include_in_schema=False)
async def static_files(
    rest_of_path: str,
):
    return FileResponse(f"app/static/{rest_of_path}")
```

**Reply by Kludex** [reply]
> very stupid question.

Not needed.

**Reply by Kludex** [reply]
@WilliamStam emailed me explaining that he meant that "very stupid question" was about his next question on the paragraph, and not about the OP's question.

Thanks for letting me know. 🙏
